graphics/src/text/paragraph.rs

```diff
@@ -251,6 +253,59 @@ impl core::text::Paragraph for Paragraph {
         Some(glyph.metadata)
     }
 
+    fn span_bounds(&self) -> HashMap<usize, Vec<Rectangle>> {
+        let internal = self.internal();
+
+        let mut all_span_bounds = HashMap::new();
```

Comment:
I originally tried `Vec<Vec<Rectangle>>` but it's possible for a span / metadata to be missing from the layout run (spaces omitted due to wrapping) which messes up the length / order of the array.